lib/SILOptimizer/Mandatory/LowerHopToActor.cpp

```diff
@@ -222,12 +208,19 @@ SILValue LowerHopToActor::emitGetExecutor(SILBuilderWithScope &B,
 
     auto subs = SubstitutionMap::get(req->getGenericSignature(),
                                      {actorType}, {actorConf});
-    auto fnType = F->getModule().Types.getConstantFunctionType(*F, fn);
 
-    auto witness =
-      B.createWitnessMethod(loc, actorType, actorConf, fn,
-                            SILType::getPrimitiveObjectType(fnType));
-    auto witnessCall = B.createApply(loc, witness, subs, {actor});
+    // Find the unwrap function
+    FuncDecl *funcDecl = ctx.getGetUnwrapLocalDistributedActorUnownedExecutor();
+    assert(funcDecl);
```

Comment:
You are likely to end up having this code run against a standard library that doesn't have this function declaration, because it's older. Can we fail gracefully in such cases?

More urgently, doesn't the use of this new function break back-deployment for distributed actors?

Reply:
Discussed this and I believe we should address this a little bit, the proposed solution is here: https://github.com/apple/swift/pull/64800

It specifically is about resilient "old" distributed actors being potentially marked as non default while they should still be default

Reply:
Done in https://github.com/apple/swift/pull/64800/ and cherry picked into this PR as well.